addons/sale/models/sale_analytic.py

```diff
@@ -90,11 +90,8 @@ def _get_sale_order_line(self, vals=None):
         result = dict(vals or {})
         so_line = result.get('so_line', False) or self.so_line
         if not so_line and self.account_id and self.product_id and (self.product_id.expense_policy != 'no'):
-            order = self.env['sale.order'].search([('project_id', '=', self.account_id.id)], limit=1)
-            if order and order.state != 'sale':
-                raise UserError(_('The Sale Order %s linked to the Analytic Account must be validated before registering expenses.') % order.name)
-
-            order = self.env['sale.order'].search([('project_id', '=', self.account_id.id), ('state', '=', 'sale')], limit=1)
+            order_in_sale = self.env['sale.order'].search([('project_id', '=', self.account_id.id), ('state', '=', 'sale')], limit=1)
+            order = order_in_sale or self.env['sale.order'].search([('project_id', '=', self.account_id.id)], limit=1) or order
```

Comment:
why this double search? the result is always the same after code traversal anyway